src/core/expr/fexpr_sumprod.cc

```diff
@@ -83,27 +83,26 @@ class FExpr_SumProd : public FExpr_Func {
         case SType::INT16:
         case SType::INT32:
         case SType::INT64:
-          return make<int64_t>(std::move(col), SType::INT64, gby, is_grouped);
+          return make<int64_t>(std::move(col), gby, is_grouped);
         case SType::FLOAT32:
-          return make<float>(std::move(col), SType::FLOAT32, gby, is_grouped);
+          return make<float>(std::move(col), gby, is_grouped);
         case SType::FLOAT64:
-          return make<double>(std::move(col), SType::FLOAT64, gby, is_grouped);
+          return make<double>(std::move(col), gby, is_grouped);
```

Comment:
Before we set the output stype in this `switch` and then casted the column to this type as 
```
col.cast_inplace(stype);
```

Now, you removed the stypes, so how the code will create `int64` result column for, let's say, `int8` input?

Reply:
The int64 is created in the sumprod.h column

Reply:
`template <typename T_IN, bool SUM, bool IS_GROUPED>
class SumProd_ColumnImpl : public ReduceUnary_ColumnImpl<T_IN, T_IN, IS_GROUPED> {
  public:
    using ReduceUnary_ColumnImpl<T_IN, T_IN, IS_GROUPED>::ReduceUnary_ColumnImpl;

    bool get_element(size_t i, T_IN* out) const override {
      T_IN result = !SUM; // 0 for `sum()` and 1 for `prod()`
      T_IN value;
      size_t i0, i1;
      this->gby_.get_group(i, &i0, &i1);
`

The int64 is passed here

Reply:
I think there is an error here and that's why the build is failing on AppVeyor.

There are two different things involved: the column's stype and the column's element type. What we did before was
- based on the column's stype determine the resulting column's stype;
- cast the input column to the resulting column's stype;
- get elements based on the resulting column's element type.

What you're doing now
- eliminated the resulting column stype;
- not casting the initial column to the resulting column stype;
- trying to get elements based on the resulting column element type.

The latter will not work, because the input column's `get_element()` could only be safely used for the input column element type.

What was the reason you removed passing the stype and casting?

Reply:
ahh thanks for the explanation. if that's the case, then i can just use `<int8_t, int64_t>` then ... that way `get_element` gets the value based on `int8`, which is the dtype of the input column, while the output column is created as `int64`.

I didnt think the column casting was necessary; however, with your explanation I have a better understanding - i'll update the code for sumprod, minmax, mean

Reply:
Yes, you can just revert your changes introduced to that switch, because we indeed need all the details like column value type, column stype, etc.